assets/scripts/streets/StreetName.jsx

```diff
@@ -58,33 +110,40 @@ class StreetName extends React.PureComponent {
 
   renderHoverPrompt = () => {
     if (this.props.isStreetReadOnly || !this.props.isHoverable) return null
-    if (typeof this.props.onClick === 'function' && this.state.isHovered) {
+    if (typeof this.props.handleStreetNameChange === 'function' && this.state.isHovered) {
       return (
         <div className="street-name-hover-prompt">
           {t('street.rename', 'Click to rename')}
         </div>
       )
     }
-
     return null
   }
 
   render () {
     let classString = 'street-name-text ' + (!needsUnicodeFont(this.props.name) ? '' : 'fallback-unicode-font')
-
-    return (
-      <div
+    if (!this.state.modify) {
```

Comment:
This `if` statement can be rewritten so that the conditions are flipped.

```js
if (this.props.editable && this.state.modify) {
   /* render editable state */
} else {
   /* render normal view */
}

This makes the logic flow much clearer to understand.

Note that I put the `props.editable` condition first because this is the most important condition. If `props.editable` is never `true`, it does not matter what this value `state.modify` is.